shared/src/test/diff/codegen/ConstructorStmt.mls

```diff
@@ -0,0 +1,84 @@
+:NewParser
+:NewDefs
+
+:js
+module Test0 {
+  log("Hello!")
+}
+//│ module Test0()
+//│ // Prelude
+//│ function log(x) {
+//│   return console.info(x);
+//│ }
+//│ let res;
+//│ let typing_unit = {
+//│   cache: {},
+//│   get Test0() {
+//│     if (this.cache.Test0 === undefined) {
+//│       class Test0 {
+//│         constructor() {
+//│           log("Hello!");
+//│         }
+//│       }
+//│       this.cache.Test0 = new Test0();
+//│       this.cache.Test0["class"] = Test0;
+//│     }
+//│     return this.cache.Test0;
+//│   }
+//│ };
+//│ globalThis.Test0 = typing_unit.Test0;
+//│ // End of generated code
+
+:js
+Test0
```

Comment:
You should try it another time to make sure it doesn't output anything the second time around!